GameDataParser/Parsers/MapEntityParser.cs

```diff
@@ -100,386 +68,345 @@ protected override List<MapEntityMetadata> Parse()
+                            InteractObjectType.Unknown, interact.interactID));
+                        break;
+                    case ISpawnPoint spawn:
+                        switch (spawn)
                         {
-                            portalType = byte.Parse(portalTypeNode?.FirstChild.Attributes["value"].Value);
-                        }
-                        string positionValue = coordNode?.FirstChild.Attributes["value"].Value ?? "0, 0, 0";
-                        string rotationValue = rotationNode?.FirstChild.Attributes["value"].Value ?? "0, 0, 0";
+                            case IEventSpawnPointNPC eventSpawnNpc: // trigger mob/npc spawns
+                                                                    // TODO: Parse "value" from NPCList.
+                                List<string> npcIds = new List<string>();
+                                npcIds.AddRange(eventSpawnNpc.NpcList.Keys);
+
+                                metadata.EventNpcSpawnPoints.Add(new MapEventNpcSpawnPoint(eventSpawnNpc.SpawnPointID, eventSpawnNpc.NpcCount, npcIds, eventSpawnNpc.SpawnAnimation, eventSpawnNpc.SpawnRadius,
+                                    CoordF.FromVector3(eventSpawnNpc.Position), CoordF.FromVector3(eventSpawnNpc.Rotation)));
+                                break;
+                            case ISpawnPointPC pcSpawn:
+                                metadata.PlayerSpawns.Add(
+                                    new MapPlayerSpawn(ToCoordS(pcSpawn.Position), ToCoordS(pcSpawn.Rotation)));
+                                break;
+                            case ISpawnPointNPC npcSpawn:
+                                // These tend to be vendors, shops, etc.
+                                // If the name tag begins with SpawnPointNPC, I think these are mob spawn locations. Skipping these.
+                                string npcIdStr = npcSpawn.NpcList.FirstOrDefault().Key ?? "0";
+                                if (npcIdStr == "0" || !int.TryParse(npcIdStr, out int npcId))
+                                {
+                                    continue;
+                                }
 
-                        MapPortalFlag flags = visibleValue ? MapPortalFlag.Visible : MapPortalFlag.None;
-                        flags |= enabledValue ? MapPortalFlag.Enabled : MapPortalFlag.None;
-                        flags |= minimapVisibleValue ? MapPortalFlag.MinimapVisible : MapPortalFlag.None;
+                                MapNpc npc = new MapNpc(npcId, npcSpawn.ModelName, npcSpawn.EntityName, ToCoordS(npcSpawn.Position),
+                                    ToCoordS(npcSpawn.Rotation));
+                                // Parse some additional flat supplemented data about this NPC.
+                                npc.IsSpawnOnFieldCreate = npcSpawn.IsSpawnOnFieldCreate;
+                                npc.IsDayDie = npcSpawn.dayDie;
+                                npc.IsNightDie = npcSpawn.nightDie;
```

Comment:
can't you just add these values to the constructor?

Reply:
In my opinion, yes.

Reply:
changed it to that.